test/integration/preflight/preflight_test.go

```diff
@@ -2,14 +2,47 @@ package preflight
 
 import (
 	"bytes"
+	"flag"
+	"fmt"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
+	"sync"
 	"syscall"
 	"testing"
 )
 
+var prebuildPlays = []string{
+	"check-yum-update/build.yml",
+}
+
+// TestMain ensure that the necessary build steps are performed
+// (in parallel) prior to running the tests.
+func TestMain(m *testing.M) {
```

Comment:
We just have one set of tests now, and they depend on the check-yum-update image, but I think the general idea for this is not to have several plays that run before ANY test, but rather a group of tests.

Think about what happens when we run:

```
go test ./... -run=Ping
```

Even though we just want to run a dummy ping test, we end up building an image that we don't need in the test.

This would be a nice fit for subtests (Go 1.7). I'm using 1.7 for quite some time now, but it is not in Fedora, CentOS and RHEL repositories yet.

https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks